unblob/extractor.py

```diff
@@ -95,11 +110,20 @@ def carve_unknown_chunks(
 
 
 def extract_valid_chunk(
-    extract_dir: Path, file: io.BufferedIOBase, chunk: ValidChunk
-) -> Path:
+    extract_dir: Path, file: io.BufferedIOBase, chunk: ValidChunk, error_queue: Queue
```

Comment:
The function name is lying. It doesn't always extract a ValidChunk, which is a code smell. Either the function is doing too much, or the name should be changed. Maybe this should be split up to 3 functions or called differently?